duration.go

```diff
@@ -122,33 +122,88 @@ func Parse(d string) (*Duration, error) {
 	return duration, nil
 }
 
+// Format formats the given duration into a ISO 8601 duration string (e.g. P1DT6H5M)
+func Format(d time.Duration) string {
```

Comment:
There should be a middle step here. Allow users to convert a `time.Duration` to our `*Duration`. Then they can call `String()` on that if they want the ISO8601 string.

Reply:
We should note in the comment that this conversion is fuzzy just like the other.